app/main.py

```diff
@@ -1,3 +1,39 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+
+    fuel_price = data["FUEL_PRICE"]
+    customers = [Customer(customer) for customer in data["customers"]]
+    shops = [Shop(shop) for shop in data["shops"]]
+
+    for customer in customers:
+        trip_costs = []
+        for shop in shops:
+            trip_cost = customer.calculate_trip_cost(shop, fuel_price)
+            trip_costs.append((shop, trip_cost))
+
+        trip_costs.sort(key=lambda x: x[1])
+
+        if trip_costs and customer.money >= trip_costs[0][1]:
+            print(f"\n{customer.name} has {customer.money} dollars")
```

Comment:
When you just start to print information about the first customer you will have the empty line and only after that, for example, "Bob has 55 dollars". It's better to find out another way to put the "\n" which will not break the output about the first costumer